bin/run-lint

```diff
@@ -1,21 +1,44 @@
 #!/usr/bin/env node
-const child_process = require( 'child_process' );
+
 const path = require( 'path' );
-var args = [ '--cache', '--quiet', '--ext=.js',  '--ext=.jsx' ];
+const child_process = require( 'child_process' );
+
+var args = [ '--cache', '--quiet', '--ext=.js,.jsx', '--format=json' ];
 if ( process.argv.length > 2 ) {
 	args = args.concat( process.argv.slice( 2 ) );
 } else {
-	args = args.concat( [ '.' ] );
+	process.stdout.write( 'No files to lint\n' );
+	process.exit( 0 );
 }
 
-const results = child_process.spawnSync( path.join( '.', 'node_modules', '.bin', 'eslint' ), args );
+const eslint = child_process.spawn( path.join( '.', 'node_modules', '.bin', 'eslint' ), args );
+const eslines = child_process.spawn( path.join( '.', 'node_modules', '.bin', 'eslines' ) );
 
-if ( results.stdout ) {
-	process.stdout.write( results.stdout );
-}
-if ( results.stderr ) {
-	process.stderr.write( results.stderr );
-}
-process.on( 'exit', function() {
-	process.exit( results.status );
-} );
+eslint.stdout.on( 'data', ( data ) => {
+	eslines.stdin.write( data );
+});
```

Comment:
Also: AFAIK, this was a script that originally only run in CircleCI and in dev machines and maybe and the time `ES6` wasn`t available for all those environments. Maybe that's the reason we do not lint the `/bin` directory. Just guessing. 

It is not a blocker, I'd prefer solving (and testing) this in another PR after this has been landed.